### DISCUSSION TITLE
Options menu doesn't work with Greasemonkey (Firefox) but works with Tampermonkey?

### DISCUSSION THREAD

**Discussion by catharsis71** [Rudokhvist/ASF-STM#39] [General]
The options menu never worked for me, using Firefox with Greasemonkey, I could open the options menu but trying to save any changes would just silently fail.  Did some experimenting today, first tried Chromium (Edge) with Tampermonkey and the script didn't work at all, couldn't even see the scan button or the options button.  Went back to Firefox, disabled Greasemonkey, installed Tampermonkey, and now it works properly.

Is this expected behavior?

**Comment by Rudokhvist** [comment]
I mainly target tampermonkey, but when possible I try to keep it compatible with other extensions, so I'll check it and see what can be done, when I have some free time.

**Comment by iBreakEverything** [comment]
Tested on Firefox 136.0.1 (64-bit) with Greasemonkey 4.13. The behavior is as described. The second confirmation would not open. I can't see  the script in Debugger tab and trying to access it from console messages would return a not found error, so I can't take a poke at it, but it might be related to unsafeWindow somehow or just how Greasemonkey works.
